### PR TITLE
Collect timing metrics for compile requests

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
I want to log some timing info for compile requests to see how many fast compiles we get vs long 20s compiles. I'm concerned prometheus may introduce an averaging effect on the data so logging seems like a decent temporary solution.

### CODE REVIEW

File: lib/base-compiler.ts [hunk 1]
```diff
@@ -2234,7 +2234,11 @@ export class BaseCompiler implements ICompiler {
                 result.retreivedFromCache = true;
                 if (doExecute) {
                     result.execResult = await this.env.enqueue(async () => {
-                        return this.handleExecution(key, executeParameters);
+                        // TODO(jeremy-rifkin) temporary metrics collection
+                        const start = performance.now();
+                        const res = this.handleExecution(key, executeParameters);
+                        logger.info(`(compilation wall clock time: ${Math.round(performance.now() - start)} ms)`);
```
Comment (mattgodbolt) [inline lib/base-compiler.ts:2240]: What's the plan for this? How will we aggregate these stats? cc @apmorton who has some fu- maybe there's an appropriate prom stat we can use (I know we tried the tracing and it was too mcuh; but some kind of aggregate min/max/median thing?)

Comment (mattgodbolt) [reply lib/base-compiler.ts:2240]: > prometheus may introduce an averaging effect on the data so logging seems like a decent temporary solution

agreed. I think there's a histogram thing though that can collect n-tiles (maybe?)

Comment (jeremy-rifkin) [reply lib/base-compiler.ts:2240]: I was planning on downloading the raw logs from papertrail and parsing the times out with a script. Super temporary thing just to collect a couple day's worth of data.

**Comment by mattgodbolt** [comment]
We spoke on Discord: will merge to get info, but will back it out by the weekend and consider using `Histogram`s in prometheus.

**Review by mattgodbolt — APPROVED** [review]

**Pull request merged**